packages/cactus-plugin-satp-hermes/src/main/typescript/plugin-satp-hermes-gateway.ts

```diff
@@ -649,4 +650,29 @@ export class SATPGateway implements IPluginWebService, ICactusPlugin {
       this.logger.warn("Server is not running.");
     }
   }
+
+  /**
+   * @notice Verify the state of the sessions before shutting down the server.
+   * This method is called before the server is shut down and awaits ensure that 
+   * all sessions are concluded before the server is terminated.
+  */
+  private async verifySessionsState(): Promise<void> {
+    const fnTag = `${this.className}#verifySessionsState()`;
+    this.logger.trace(`Entering ${fnTag}`);
+    if (!this.BLODispatcher) {
+      throw new Error(`Cannot ${fnTag} because BLODispatcher is erroneous`);
+    }
+    this.BLODispatcher.setInitiateShutdown();
+    const manager = await this.BLODispatcher.getManager();
+    let status = false;
+    while (!status) {
```

Comment:
there is a potential infinite loop. this function might block, and we need to decide if this is desirable behaviour. On one hand, the gateway cannot handle anything else; on the other, it should not serve new requests. So there should be a middle-term that allows the gateway to keep serving other requests (namely admin) while not initiating new transfers nor blocking.

Besides, we might want to include a timeout, since something can be broken in the manager (in particular if there is a hanging SATP transfer, communicating with a counterparty gateway).